auth-agents-common/src/main/java/org/apache/atlas/policytransformer/PersonaCachePolicyTransformer.java

```
@@ -88,44 +99,59 @@ public List<AtlasEntityHeader> transform(AtlasEntityHeader atlasPolicy) {
                 header.setAttribute(ATTR_POLICY_ACTIONS, templatePolicy.getActions());
                 header.setAttribute(ATTR_POLICY_RESOURCES_CATEGORY, templatePolicy.getPolicyResourceCategory());
                 header.setAttribute(ATTR_POLICY_IS_ENABLED, getIsPolicyEnabled(atlasPolicy));
+                header.setAttribute(ATTR_NAME, "transformed_policy_persona");
+                header.setAttribute(ATTR_POLICY_CONDITIONS, templatePolicy.getPolicyConditions());
 
-                String subCategory = getPolicySubCategory(atlasPolicy);
-
-                List<String> finalResources = new ArrayList<>();
-
-                for (String templateResource : templatePolicy.getResources()) {
-                    if (templateResource.contains(PLACEHOLDER_ENTITY)) {
-                        for (String entityResource : entityResources) {
-                            finalResources.add(templateResource.replace(PLACEHOLDER_ENTITY, entityResource));
+                if (policyServiceName.equals(POLICY_SERVICE_NAME_ABAC)) {
+                    if (policyFilterCriteria != null && !policyFilterCriteria.isEmpty()) {
+                        ObjectMapper mapper = new ObjectMapper();
+                        try {
+                            JsonNode filterCriteriaNode = mapper.readTree(policyFilterCriteria);
+                            if (filterCriteriaNode != null && filterCriteriaNode.get("entity") != null) {
+                                JsonNode entityFilterCriteriaNode = filterCriteriaNode.get("entity");
+                                policyFilterCriteria = entityFilterCriteriaNode.toString();
+                            }
+                        } catch (JsonProcessingException e) {
+                            e.printStackTrace();
```

this just swallows the exception ? no logging or rethrow ?